fca/src/main/java/org/nmdp/ngs/fca/CrossTable.java

```diff
@@ -0,0 +1,270 @@
+/*
+
+    ngs-fca  Formal concept analysis for genomics.
+    Copyright (c) 2014-2015 National Marrow Donor Program (NMDP)
+
+    This library is free software; you can redistribute it and/or modify it
+    under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or (at
+    your option) any later version.
+
+    This library is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library;  if not, write to the Free Software Foundation,
+    Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA.
+
+    > http://www.gnu.org/licenses/lgpl.html
+
+*/
+package org.nmdp.ngs.fca;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import java.util.List;
+import java.util.Iterator;
+import java.util.ArrayList;
+
+import org.dishevelled.bitset.MutableBitSet;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.common.collect.Range;
+import com.tinkerpop.blueprints.Graph;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import org.dishevelled.bitset.AbstractBitSet;
+
+/**
+ * CrossTable.
+ * @param <G> object type
+ * @param <M> attribute type
+ */
+public final class CrossTable implements Iterable<CrossTable.Row> {
```

Comment:
I looked into adding ```viewRow(long)/viewColumn(long)``` style methods to BitMatrix2D as I mentioned and it looks like too much work.  Using an implementation of ```Matrix2D<Boolean>``` from the same library would provide that functionality but of course at a much worse memory footprint.